runtime_scan/pkg/provider/gcp/client.go

```diff
@@ -0,0 +1,519 @@
+		"ScannerZone":  c.gcpConfig.ScannerZone,
+		"Provider":     string(c.Kind()),
+	})
+
+	err := c.ensureScannerVirtualMachineDeleted(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to ensure scanner virtual machine deleted: %w", err)
+	}
+	logger.Debugf("Deleted scanner virtual machine")
+
+	err = c.ensureTargetDiskDeleted(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to ensure target disk deleted: %w", err)
+	}
+	logger.Debugf("Deleted disk")
+
+	err = c.ensureSnapshotDeleted(ctx, config)
+	if err != nil {
+		return fmt.Errorf("failed to ensure snapshot deleted: %w", err)
+	}
+	logger.Debugf("Deleted snapshot")
+
+	return nil
+}
+
+func (c *Client) DiscoverScopes(ctx context.Context) (*models.Scopes, error) {
+	var ret models.Scopes
+	ret.ScopeInfo = &models.ScopeType{}
+	var regions []models.GcpRegion
+
+	gcpRegions, err := c.listAllRegions(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("failed to list regions: %w", err)
+	}
+
+	for _, region := range gcpRegions {
+		regions = append(regions, models.GcpRegion{
+			Name:  *region.Name,
+			Zones: utils.PointerTo(getZonesLastPart(region.Zones)),
+		})
+	}
+
+	err = ret.ScopeInfo.FromGcpProjectScope(models.GcpProjectScope{
+		ProjectID: c.gcpConfig.ProjectID,
+		Regions:   &regions,
+	})
+	if err != nil {
+		return nil, provider.FatalErrorf("failed to convert from gcp project scope: %v", err)
+	}
+
+	return &ret, nil
+}
+
+// nolint: cyclop
+func (c *Client) DiscoverTargets(ctx context.Context, scanScope *models.ScanScopeType) ([]models.TargetType, error) {
+	var ret []models.TargetType
+
+	gcpScanScope, err := scanScope.AsGcpScanScope()
+	if err != nil {
+		return nil, fmt.Errorf("failed to convert as gcp scan scope: %v", err)
+	}
+
+	// get list of zones to scan
+	zones, err := c.getZonesToScanFromScanScope(ctx, gcpScanScope)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get zones to scan from scan scope: %w", err)
+	}
+
+	// prepare include and exclude tags filter
+	//filter := prepareTagsFilter(gcpScanScope.InstanceTagSelector, gcpScanScope.InstanceTagExclusion)
+
+	// TODO (erezf) unfortunately the tags filter is broken in the google api, so I will need to fetch all instances in required zones, and filter by tags in memory.
+	for _, zone := range zones {
+		targets, err := c.listInstances(ctx, nil, zone, gcpScanScope)
+		if err != nil {
+			return nil, fmt.Errorf("failed to list instances: %w", err)
+		}
+
+		ret = append(ret, targets...)
+	}
+
+	return ret, nil
+}
+
+// [https://www.googleapis.com/compute/v1/projects/gcp-etigcp-nprd-12855/zones/us-central1-c, https://www.googleapis.com/compute/v1/projects/gcp-etigcp-nprd-12855/zones/us-central1-a] -> [us-central1-c, us-central1-a]
+func getZonesLastPart(zones []string) []string {
+	var ret []string
+
+	for _, zone := range zones {
+		ret = append(ret, getLastURLPart(&zone))
+	}
+	return ret
+}
+
+func getInstanceBootDisk(vm *computepb.Instance) (*computepb.AttachedDisk, error) {
```

Comment:
There can be only one boot disk AFAIK

Reply:
But it's a slice :)

Reply:
vm.Disks is a slice that is true. But the boolean "Boot" can only be for one disk. The other disks should have this boolean false and are attached disks. Otherwise not sure how the the vm will know from which disk to boot